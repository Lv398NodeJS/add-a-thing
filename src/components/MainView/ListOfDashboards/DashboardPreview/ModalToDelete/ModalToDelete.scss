.delete-message {
  margin: 0;
}