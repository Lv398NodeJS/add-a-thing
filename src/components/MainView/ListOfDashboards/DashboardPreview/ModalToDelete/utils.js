import db from '../../../../../fire';

export const deleteLocallyAndRemotely = (
  id, deleteDashboard, dashboards,
) => {
  delete dashboards.id;
  deleteDashboard(dashboards);
  const dashboardRef = db.database().ref(`/dashboards/${id}`);
  dashboardRef.remove();
};
