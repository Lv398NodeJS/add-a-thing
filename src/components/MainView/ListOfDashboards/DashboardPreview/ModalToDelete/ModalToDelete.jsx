import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as viewActions from '../../../../../actions/mainViewActions';
import './ModalToDelete.scss';
import { deleteLocallyAndRemotely } from './utils';

export class ModalToDelete extends Component {
  constructor(props) {
    super(props);

    this.state = { show: props.show };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ show: nextProps.show });
  }

  handleClose = () => {
    const { toggleModal } = this.props;
    toggleModal();
    this.setState({ show: false });
  }

  handleConfirmDelete = () => {
    const { showComponent } = this.state;
    const { mainViewActions, id, dashboards } = this.props;
    deleteLocallyAndRemotely(id, mainViewActions.deleteDashboard, dashboards);
    this.setState({
      showComponent: !showComponent,
    });
  }

  render() {
    const { show } = this.state;
    return (
      <Modal
        onHide={this.handleClose}
        show={show}
        className="confirm-delete-modal"
      >
        <Modal.Dialog className="delete-message">
          <Modal.Header closeButton>
            <Modal.Title>Are you sure you want to delete this dashboard?</Modal.Title>
          </Modal.Header>

          <Modal.Body>
            <p>This action cannot be undone.</p>
          </Modal.Body>

          <Modal.Footer>
            <Button
              className="cancel-btn"
              variant="secondary"
              onClick={this.handleClose}
            >
            Cancel
            </Button>
            <Button
              variant="outline-danger"
              onClick={this.handleConfirmDelete}
            >
              Delete this dashboard
            </Button>
          </Modal.Footer>
        </Modal.Dialog>
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  dashboards: state.mainViewReducer.dashboards,
});

const mapDispatchToProps = dispatch => ({
  mainViewActions: bindActionCreators(viewActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalToDelete);
