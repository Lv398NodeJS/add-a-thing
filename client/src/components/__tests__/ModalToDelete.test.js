import React from 'react';
import { shallow } from 'enzyme';
import { ModalToDelete } from '../MainView/ListOfDashboards/DashboardPreview/ModalToDelete/ModalToDelete';

describe('ModalToDelete', () => {
  const component = shallow(<ModalToDelete />);
  it('should render correctly in "debug" mode', () => {
    const componentDebug = shallow(<ModalToDelete debug />);

    expect(componentDebug).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    expect(component).toMatchSnapshot();
  });

  it("closes modal window when 'Cancel' button clicked", () => {
    const shallowWithProps = shallow(<ModalToDelete toggleModal={jest.fn()} />);

    shallowWithProps.setState({ show: true });
    shallowWithProps.find('.cancel-btn').simulate('click');
    shallowWithProps.update();

    expect(shallowWithProps.state().show).toEqual(false);
  });
});
