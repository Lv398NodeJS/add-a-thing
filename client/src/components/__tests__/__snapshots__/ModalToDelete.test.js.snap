// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalToDelete should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`ModalToDelete should render correctly with no props 1`] = `ShallowWrapper {}`;
